cmd/policy.go

```diff
@@ -80,14 +80,23 @@ func (sys *PolicySys) IsAllowed(args policy.Args) bool {
 	sys.RLock()
 	defer sys.RUnlock()
 
-	// If policy is available for given bucket, check the policy.
-	if p, found := sys.bucketPolicyMap[args.BucketName]; found {
-		return p.IsAllowed(args)
+	isPolicyAllowed := func() bool {
+		// If policy is available for given bucket, check the policy.
+		if p, found := sys.bucketPolicyMap[args.BucketName]; found {
+			return p.IsAllowed(args)
+		}
+
+		// As policy is not available for given bucket name, returns IsOwner i.e.
+		// operation is allowed only for owner.
+		return args.IsOwner
+	}
+
+	// If opa is configured, let the policy arrive from Opa
+	if globalOpaPolicySys != nil {
+		return globalOpaPolicySys.IsAllowed(args) && isPolicyAllowed()
 	}
 
-	// As policy is not available for given bucket name, returns IsOwner i.e.
-	// operation is allowed only for owner.
-	return args.IsOwner
+	return isPolicyAllowed()
```

Comment:
simpler:
```
        // If opa is configured, let the policy arrive from Opa
        if globalOpaPolicySys != nil && !globalOpaPolicySys.IsAllowed(args) {
                return false
        }

        // If policy is available for given bucket, check the policy.
        if p, found := sys.bucketPolicyMap[args.BucketName]; found {
                return p.IsAllowed(args)
        }

        // As policy is not available for given bucket name, returns IsOwner i.e.
        // operation is allowed only for owner.
        return args.IsOwner
```

Reply:
We want to honor bucket policies along with OPA policies if they are set @krishnasrinivas , this code is only applicable if we disallow bucket policy calls if OPA is configured we need to figure out if this correct approach.

Reply:
yes @harshavardhana the code I posted does the same thing as your code